### ISSUE TITLE
missing email from SABilling system

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#399]
In GitLab by @kdjstudios on Jun 16, 2017, 10:32

Charlie wrote: http://www.servicedesk.answernet.com/profiles/ticket/2017-06-15-42113/conversation

I just realized this last billing period that at the time of billing, we are no longer receiving the email from SA that give the list of emailed invoices sent out.

The last one we received was from the 1/3/17 billing period. That was a very helpful piece of info to track to see who all we send invoices by email. And if any bounce back.

I clarified and Charlie confirmed:

After you complete your billing cycle, you are then clicking the "Send Emails" button; which will send out email invoices appropriately. This should then as well send a confirmation email to "billing@towneanswering.com" which is listed under the site. But you have not received the confirmation email since January Billing?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 16, 2017, 10:35

@smasih @tim.traylor 

I think this was happening on an internal site before, but it was a user error in them forgetting to 'click send emails'. Which is why I confirmed with Charlie. Could we please look into this as this has not been working for a few months now.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 19, 2017, 08:36

@smasih @abakshi @tim.traylor 

Has there been any progress on this?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 19, 2017, 08:55

@kdjstudios @abakshi 

After testing, I can confirm that **send email** is working from code point of view, there should be some other issue. @tim.traylor can help us in this and will give you more update on this tomorrow.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 19, 2017, 09:11

@smasih @kdjstudios Hi Kyle or Sumeet, if either of you can tell me what the subject lines of the emails would be and who it would have been sent to, I can have Robert look at the email server log and see if they are being sent out and possibly blocked by Charlie?

Thx,
Tim

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 19, 2017, 09:26

@tim.traylor 

I believe the subject line should be along the lines of this: Emailed XX Invoices for XXXX Site XX/XX/XXXX Master

Example; I just ran though the Demo site Billing cycle, and the one I received was: Emailed 0 Invoices for Demo Site 06/25/2017 Master

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 20, 2017, 01:35

@tim.traylor 

The subject line of the email is:   
**Towne Answering Service Invoice for account _\< Acc # >_**

In my testing this morning, I updated my email ID on one of the Towne Account and mailed an Invoice to myself through sab. As a result, I received the Invoice. (*later I restored the email address*)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 20, 2017, 14:02

@smasih @tim.traylor 

Update:

OK, we are doing billing, we emailed 75 invoices from SA, we did not receive the confirmation email from the SAB system showing who was emailed.

I'm getting some auto-responses from clients mail systems, so that's the only way I know it might be working.

My guess is that the support team did nothing.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 20, 2017, 14:43

@smasih 

This is not that individual account invoices are not being emailed. It is that the summary email sent to the site's email address is not being delivered after the site has clicked the sent emails button.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 20, 2017, 14:44

@smasih @tim.traylor 

I looked at their last billing cycle and it appears possibly that Charlie has not clicked the 'send emails' button because the revert button is still visible; or is it because I am an admin user?

![Untitled](/uploads/58c1666abbc46a523bcbe2e9f6fd97f4/Untitled.png)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 21, 2017, 02:03

@kdjstudios 

* First of all, please help me to understand that what exactly is the issue with Keener Site..?
  * Emails are not being sent
  * Garret is not receiving an confirmation email on **Billing@Keenercom.Net** (*probably because email address is capitalized*)
  * There is no confirmation text like ..  ![Screenshot](/uploads/4ef36b9af226b29929ac6146e67529f3/Screenshot.png)

Secondly, code is working on Hosted I checked with testy site. 

Lastly, functionality for hiding the **Revert** button on click of send email is not live on Hosted.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 21, 2017, 10:00

@smasih 

As previously stated Garrett is not receiving any confirmation email after sending invoices. 
What do you mean the email is capitalized? And why would that have any effect on sending an email to it?

I posted screen shots below: I do not see any capitalization?

![Untitled2](/uploads/6580b57569cc8e617fee912b5d7b242b/Untitled2.png)![Untitled](/uploads/6371c3b2d478adc03cb1ea71be804343/Untitled.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 21, 2017, 10:03

@smasih @tim.traylor 

Why would the functionality for hiding the revert button not be live on Hosted? Was it not part of the a previous release schedule?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 21, 2017, 10:10

@smasih 

I think you may be looking at the wrong site. This is for Towne Answering Service not Keener Communications.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 21, 2017, 14:51

I have also sent a test email to billing@towneanswering.com and received confirmation that it is receiving emails.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 22, 2017, 08:30

@smasih @tim.traylor 

It has been over 4 months that this has been reported as non working. Increasing the priority to find a resolution.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 23, 2017, 08:45

@kdjstudios @tim.traylor 

I got confused Garret is Keener and Crown is Towne, and you have been mentioning Garret in all your comments.

Tomorrow I will update all accounts with our test email and my email on Site level to receive a confirmation email. If this works means it means code is fine and we need to check on SMTP level/Email service logs if we are sending the emails on  "billing@towneanswering.com"

Lastly,  Towne needs to look into their email folders other than inbox, can't think of anything else.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 23, 2017, 08:51

@smasih @tim.traylor 

Please as Tim stated, if something does not make sense to ask, not assume. Please use the Whatsapp as a resource.  

I have already asked Charlie and Deb to look into their spam folders and other items, they were unable to find any notices.

@tim.traylor 

Were you and Robert able to find any information in the email logs?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 23, 2017, 09:03

@kdjstudios 

I have done the same with **Testy Site** and everything went fine.

**All the site uses same base code, but still we will give it a shot tomorrow**

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 24, 2017, 03:26

@kdjstudios @tim.traylor 

Tested with keener and towne... 
* Updated @pchaudhary 's email on all accounts and my email on site level (to receive confirmation email)
* We received all emails
* along with confirmation email, however they landed in our spam folder
* restored the db with original emails.

**After testing I can confirm that there is no issue with code, so now I believe we should take the next  step by checking Answernet's email provider/smtp server reports**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 24, 2017, 08:19

@smasih 

We will need to wait for @tim.traylor and Robert to confirm they are being sent out via the email logs before marking as completed. Along with Charlie successfully receiving.

Would we be able to send out a test just like you did above with the clients email "billing@towneanswering.com"?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 26, 2017, 24:46

@kdjstudios  sure we can, just please inform them that they will be receiving a test email.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 26, 2017, 08:27

@smasih 

Thank you, I have informed charlie we will be sending out test emails over the next few days. If we could please send out an initial five test confirmation emails to 'billing@towneanswering.com' and then two test emails each to 'ccrown@towneanswering.com' and 'dcrown@towneanswsering.com' Should be enough to help us start figuring out what is preventing the confirmation email.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 26, 2017, 08:28

@tim.traylor 

Any update with Robert with the possibility of the email server blocking or marking these emails as spam?

Also any update on finding any logs of these messages being sent out for the last few months?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 27, 2017, 05:15

@kdjstudios 

We were not able to send test mails today, we will be sending it tomorrow.

moving back to testing

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 30, 2017, 14:38

@smasih @tim.traylor 

Have you sent any test emails? Charlie informed me that he has not received any.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 3, 2017, 08:53

@smasih @pchaudhary @abakshi 

Were there any test emails sent out? Charlie received no emails.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 3, 2017, 12:43

Charlie reached out to me today and said that he had found one email in his spam. I have attached what he was able to forward me below in a few different save formations. The one marked [spam] will show why is was marked as spam in a email. If you look at  this one it appears that the date and time were the major deciding factors in marking as spam.

[FW_SPAM_Emailed_72_Invoices_for_Towne_Answering_Service_05212017_Master.txt](/uploads/fd2cb300034bb16d03cb6e9fd972b211/FW_SPAM_Emailed_72_Invoices_for_Towne_Answering_Service_05212017_Master.txt)

[SPAM_Emailed_72_Invoices_for_Towne_Answering_Service_05212017_Master__12.1_KB_.msg](/uploads/f6435d60460e9e5e8a711b294ea314e0/SPAM_Emailed_72_Invoices_for_Towne_Answering_Service_05212017_Master__12.1_KB_.msg)

[FW_SPAM_Emailed_72_Invoices_for_Towne_Answering_Service_05212017_Master.msg](/uploads/24ec79494d5e3753eec698022fb3fdc4/FW_SPAM_Emailed_72_Invoices_for_Towne_Answering_Service_05212017_Master.msg)

[FW_Emailed_72_Invoices_for_Towne_Answering_Service_05212017_Master.htm](/uploads/4a63fc851d5db087dbab66ec39194484/FW_Emailed_72_Invoices_for_Towne_Answering_Service_05212017_Master.htm)

[FW_Emailed_72_Invoices_for_Towne_Answering_Service_05212017_Master.msg](/uploads/11052ec04d3b7deff1c33f8e4facb1d3/FW_Emailed_72_Invoices_for_Towne_Answering_Service_05212017_Master.msg)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 3, 2017, 12:44

Also if you could inform us an approximation on when you will be sending the emails I can update their team to be aware and more alert in checking their spam during those time frames.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jul 4, 2017, 24:48

@kdjstudios 

On 3rd we sent few test confirmation mails out on the mentioned emails. Emails were sent in morning through SAB. 

Means, Code is working, email is working....... so is there anything else we can do to help..??

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jul 5, 2017, 07:36

I have Robert looking into the issue: http://www.servicedesk.answernet.com/profiles/ticket/2017-07-05-82998/conversation

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 5, 2017, 08:57

@tim.traylor 

Thank you for having Robert look into this.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 5, 2017, 10:40

@smasih @tim.traylor 

Reply from Robert:

I checked the server and the server time/date is correct.  The email was just the SA headers and it reports the time/date was several hours in the future so it might be something is the SAB code that generated the report.
Thanks
Robert

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jul 6, 2017, 07:59

@kdjstudios 

Actually, time was in future due to the difference in time zone.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jul 6, 2017, 08:51

@smasih Hi Sumeet, Robert found the following:
"The date/time on the Envelope is set by the sending program so the app that generated the email is the one that set the time/timezone wrong.  The headers are set by the MTA In our case Postfix as it goes though the systems to it destination. "

So it looks like the maybe the gem that's creating the email isn't using or specifying a time zone.  I don't know that everyone sets that SPAM flag they used in SpamAssassin, so I think it's possible it's been doing this all along, and that a change to their settings is causing this to be a problem now.

Can you please check the gem and see what our options are?

Thanks,
Tim

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jul 8, 2017, 07:17

@kdjstudios @tim.traylor 

as per @pchaudhary's investigation ... 
* System time is correct (configured as per EST)
* On clicking **Send Emails** button on billing cycle page 
  * emails are sent to acc holder's  email add with correct time.
 * however, when a confirmation email is sent to Site holder... the time for that particular email is in-accurate.

need to do more investigation on this

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 17, 2017, 14:16

@smasih @tim.traylor 

Charlie update:

We just sent out emailed invoices from SAB, I'm getting automated email replies from customers but have no email from SABilling saying who we sent the emails out to.
I'm thinking something still doesn't work correctly.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jul 18, 2017, 24:43

@kdjstudios 

We, have tested the code couple of times... it is sending the emails and confirmation email out..!!. @tim.traylor I think we need to investigate further. 

And Kyle, I am assuming that you would have asked Charlie to check his spam/junk folder..?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 18, 2017, 08:17

@smasih @tim.traylor 

I will inform him to check his SPAM folder again, but I believe we came to the resolution that this was not an issue with the email sending, but the email sending with the wrong times; thus causing it to be marked as spam. Was the correct time ever resolved.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 18, 2017, 09:26

@smasih @tim.traylor 

Update from Charlie:

 I checked all the spam filters before I emailed you.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 18, 2017, 09:34

@smasih @tim.traylor 

Charlie also forwarded a copy of an old email to insure we are looking into the right email.

[FW_Emailed_53_Invoices_for_Towne_Answering_Service_01012017_Master.txt](/uploads/9fb45540c89bf923d7e75117bcdcb802/FW_Emailed_53_Invoices_for_Towne_Answering_Service_01012017_Master.txt)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 18, 2017, 13:52

@smasih @tim.traylor 
Another update from Charlie:

Subject: RE: missing email from SABilling system  
I checked the server logs from yesterday for the words sabilling, sahosted, and answernet.  That would give me any communication we got from servers with those phrases in the domain name.  Keep in mind that includes EVERYTHING, even stuff deleted outright as spam.  

The only thing that came up was the email that is sent to me for an accounts invoice, which I attach a report to and forward to the customer.  That was sent to me at 3:56 pm yesterday, but it is sent FROM billing@towneanswering.com, not to it.

I have also logged into the billing@ account on Intermedia, in the event that they were for some reason sending it directly there (it's our tertiary MX record) and skipping our mailservers on site here, but that account is empty.  If they can give me the IP that would be sending to us I can search the logs again.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Jul 19, 2017, 06:46

@smasih 

Please attach our logs from SABilling server and we may check the logs from the email service provider being used to send these emails. 

@tim.traylor 

Looking forward to your suggestions and assistance to identify the real cause.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jul 19, 2017, 07:28

@kdjstudios Hi Kyle, what Charlie had forwarded to us previously indicated that email from us could be being marked as SPAM by their SpamAssassin software.  The fastest resolution is to have them whitelist email from sahosted.com.  Can you please present that to them?  Thx, Tim

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 19, 2017, 08:30

@tim.traylor @abakshi @smasih @pchaudhary 

I just received this update from Nate:

Kyle,

When you send the message on our behalf, the SMTP logs show an Answernet server, so that is the easiest thing to search for since it will show anything coming from your servers, regardless of destination and sender – and if the connection is rejected outright due to spam filtering before the SMTP transaction occurs, we won’t see the rest of the info:

"SMTPD"             2168       1427336                "2017-07-17 15:56:58.559"            "104.37.108.241"              "RECEIVED: EHLO ann200mail03.answernet.com"

I did find it, it was coming through at the exact same time as one that is sent to me for a customer (so I can attach reports and forward to them).  For some reason, the header in the summary email sent to billing@towneanswering.com is different – when the email passes through our perimeter spam filtering relay, the sender/from header info is dropped.

Here is an example of the invoice I get sent to ngefvert@towneanswering.com leaving the relay (which I receive successfully):

"APPLICATION" 1940       "2017-07-17 15:56:59.828"            "SMTPDeliverer - Message 2259808: Delivering message from billing@towneanswering.com to ngefvert@towneanswering.com. File: C:\Program Files (x86)\hMailServer\Data\{967A9208-9952-4E97-9CD3-C3A011CDE395}.eml"

And here is the one sent to billing@towneanswering.com:

"APPLICATION" 2060       "2017-07-17 15:57:01.407"            "SMTPDeliverer - Message 2259812: Delivering message from <Empty> to billing@towneanswering.com. File: C:\Program Files (x86)\hMailServer\Data\{446A0D71-4259-4BFF-A523-81126698B535}.eml"

So there is something different about that message…  And with that empty from header exiting the relay, our Exchange server rejects it.

I do not know why that message is acting differently upon relay, but for the sake of getting it working I made some changes to how the relay passes through for emails to billing@towneanswering.com.  Is there a way to resend that emailed invoice report again to see if the changes fixed the issue, or do we have to wait until the next billing cycle.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 19, 2017, 08:32

Overall it appears that Nate had the same idea to whitelist the email. 

@smasih 

Could we resend their last billing cycles email invoice confirmations? If so, wonderful and please let me know once we do. Otherwise could we please send a few more test messages to confirm that their change is working. Thank you in advance.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jul 19, 2017, 08:57

@kdjstudios 

If we send the confirmation email again it will send email to all the accounts as well. So, if it's fine you can do it from your end as well from billing cycles page.  
But, if they just want the confirmation email we need to suppress the email of all the account and which is not possible at the moment due to high user traffic.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 19, 2017, 09:25

@smasih 

I will check with them on resending the all the invoices again. However this is not something I can complete as the option to send invoices is not visible. * I believe this has to do with us removing this for all but tech admins to prevent duplicates from being sent.

![Untitled](/uploads/145ec714df9090ef5a0af58c6da0b454/Untitled.png)

I would say lets proceed with your second  suggestion and suppress the email to the accounts. Please let me know when you can perform this.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jul 20, 2017, 05:02

@kdjstudios 

done,  emailed confirmation mail to **billing@towneanswering.com**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 20, 2017, 09:03

@smasih 

Thank you, Charlie confirmed that he was able to receive the email!!!!! This is awesome.

@tim.traylor 

Should we still be looking into why the email was being marked as spam (due to mismatching timezone I believe), or should we close this ticket?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Aug 19, 2017, 03:09

Hi [Kyle] ( @kdjstudios ),   
Can we close this one?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 19, 2017, 12:29

@pchaudhary 

Lets get @tim.traylor approval for this. I believe we should be able to, as it appears to mainly be an issue with the clients spam detector. But I do not know if we were able to fix anything with the outbound timezone it was using on the emails which is what was causing the spam filter to mark it.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Aug 25, 2017, 01:15

Hi [Kyle] ( @kdjstudios ),    
I think we can close this one based on our yesterday's call.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 25, 2017, 08:35

@pchaudhary 

Agreed, Marking to close.
